### DISCUSSION TITLE
What does Nuxt 3 offer that Quasar 2 doesn't?

### DISCUSSION THREAD

**Discussion by flowt-au** [quasarframework/quasar#16962] [General - Components / Directives / etc]
I have been using Quasar for SPAs. Love it! 

I need a static website (marketing / documentation / contact / about / etc ...) for one of the SPAs. So, I looked at Nuxt (which I have zero experience of). The more I look, the more I can't see what the advantage is over using Quasar. 

Is the main advantage that there are templates for Nuxt for websites, ecommerce sites and documentation sites etc ? I can see that would be helpful but wondering if that is the only advantage over coding in Quasar from the start. I see you can even use Quasar in Nuxt, which again leads me back to asking why wouldn't I just use Quasar in the first place?

I see people using headless CMS (eg Directus) and of course that can be done with Quasar too. Nothing specific to Nuxt there.

And ... if anyone has used one of those templates for a "SPA promotional / management site" are there any recommendations? (I know that is really vague but at this stage I am just looking for a general direction here).

Thanks, 
Murray

**Comment by UkonnRa** [comment]
Nuxt Runtime Config? https://nuxt.com/docs/guide/going-further/runtime-config

A common use case. My project has several environments like Production and Staging, each needs some specific configs like `API_BASE` etc.
Of course I can set them when building, but it means I need TWO docker images when deploying, like `app-ui:1.0-staging` and `app-ui:1.0-production`.

I hope to use one image, and config all the items using Env Vars, which can be achieved via SSR.

I found Nuxt can provide the ability, but I don't know if Quasar can do so. At least I cannot find any hints in the doc.

**Reply by rstoenescu** [reply]
This is entirely not true.
* q/app-vite: https://quasar.dev/quasar-cli-vite/handling-process-env#adding-to-process-env
* q/app-webpack: https://quasar.dev/quasar-cli-webpack/handling-process-env#adding-to-process-env

Moreover, env files are automatically taken into account for the new q/app-* betas:
* https://quasar.dev/quasar-cli-vite/upgrade-guide#the-env-dotfiles-support
* https://quasar.dev/quasar-cli-webpack/upgrade-guide#the-env-dotfiles-support

**Comment by flowt-au** [comment]
This short video gives an example of how Nuxt Content offers a level of abstraction for more "static" type sites.
[https://www.youtube.com/watch?v=o9e12WbKrd8&t=174s](https://www.youtube.com/watch?v=o9e12WbKrd8&t=174s)
Anyway, beginning to see where Nuxt fits in.
Murray
